Kudu.Core/Jobs/BaseJobRunner.cs

```diff
@@ -72,17 +71,65 @@ protected string WorkingDirectory
 
         protected JobSettings JobSettings { get; set; }
 
-        private static int CalculateHashForJob(string jobBinariesPath)
+        internal static bool JobDirectoryHasChanged(
+            Dictionary<string, FileInfoBase> sourceDirectoryFileMap, 
+            Dictionary<string, FileInfoBase> workingDirectoryFileMap, 
+            Dictionary<string, FileInfoBase> cachedSourceDirectoryFileMap)
         {
-            var updateDatesString = new StringBuilder();
-            DirectoryInfoBase jobBinariesDirectory = FileSystemHelpers.DirectoryInfoFromDirectoryName(jobBinariesPath);
+            // enumerate all source directory files, and compare against the files
+            // in the working directory (i.e. the cached directory)
+            FileInfoBase foundEntry = null;
+            foreach (var entry in sourceDirectoryFileMap)
+            {  
+                if (workingDirectoryFileMap.TryGetValue(entry.Key, out foundEntry))
+                {
+                    if (entry.Value.LastWriteTimeUtc != foundEntry.LastWriteTimeUtc)
```

Comment:
Indeed, as it stands that won't be handled correctly. I think we can't avoid special casing the extension here. We have a list of extensions in the code that you could use to avoid hard coding it here.